overlord/configstate/configcore/hostname.go

```diff
@@ -83,6 +82,10 @@ func handleHostnameConfiguration(_ sysconfig.Device, tr config.ConfGetter, opts
 			return fmt.Errorf("cannot set hostname: %v", osutil.OutputErr(output, err))
 		}
 	} else {
```

Comment:
The `if` at the beginning of this `else` is:
```
	if opts == nil {
```
which is prone to break if we add more options. What I would rather do is to create an empty `fsOnlyContext` structure at the beginning of this function if `opts == nil`, and change this `if` here to
```
    if opts.RootDir == "" {
```

But given the urgency I propose to land this one as-is, and I'll submit a small PR to `master` with this suggestion.

Reply:
Actually, according to how the `coreOnly` variable is initialized in `overlord/configstate/configcore/handlers.go`, this is never `nil`.

Reply:
Actually, according to how `fsOnlyContext` is initialized in https://github.com/snapcore/snapd/blob/master/overlord/configstate/configcore/handlers.go#L155-L157, `opts` is never `nil`, so we were never taking the `hostnamectl` branch.

Reply:
fsOnlyContext being non-nil is mainly used from configuring the filesystem in the initramfs, etc see the sysconfig package for examples

Also I agree this pattern is confusing but the sysconfig package is also rather confusing and I think there is a history behind why folks went with `opts == nil` to mean runtime only implementation

Reply:
@mardy the opts==nil path is taken from applyHandlers in runstate.go; as @anonymouse64 said there are two cases when config maybe applied. I'd be careful changing that and if we do, all config handlers need to be verified since this pattern is in more places.